cocoa/src/toga_cocoa/icons.py

```diff
@@ -13,25 +13,24 @@ def __init__(self, interface, path):
         self.path = path
         try:
             # We *should* be able to do a direct NSImage.alloc.init...(), but if the
-            # image file is invalid, the init fails, and returns NULL - but we've
-            # created an ObjC instance, so when the object passes out of scope, Rubicon
-            # tries to free it, which segfaults. To avoid this, we retain result of the
-            # alloc() (overriding the default Rubicon behavior of alloc), then release
-            # that reference once we're done. If the image was created successfully, we
-            # temporarily have a reference count that is 1 higher than it needs to be;
-            # if it fails, we don't end up with a stray release.
+            # image file is invalid, the init fails, returns NULL, and releases the
+            # Objective-C object. Since we've created an ObjC instance, when the object
+            # passes out of scope, Rubicon tries to free it, which segfaults.
+            # To avoid this, we retain result of the alloc() (overriding the default
+            # Rubicon behavior of alloc), then release that reference once we're done.
+            # If the image was created successfully, we temporarily have a reference
+            # count that is 1 higher than it needs to be; if it fails, we don't end up
+            # with a stray release.
             image = NSImage.alloc().retain()
             self.native = image.initWithContentsOfFile(str(path))
             if self.native is None:
                 raise ValueError(f"Unable to load icon from {path}")
         finally:
             image.release()
 
-        # Multiple icon interface instances can end up referencing the same native
-        # instance, so make sure we retain a reference count at the impl level.
-        self.native.retain()
-
     def __del__(self):
+        # Calling `release` during init disabled Rubicon's "release on delete"
+        # automation. We therefore need to release manually here.
```

Comment:
This comment doesn't makes sense to me - which "release" are you referring to?

Reply:
I've rephrased the comment and moved it to the init method instead. Hopefully its clearer now?